### PR TITLE
update to WildFly23

### PR DESCRIPTION
**Pull request by mbiarnes**
**Thank you for submitting this pull request**

[droolsjbpm-build-bootstrap](https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1626)
[kie-soup](https://github.com/kiegroup/kie-soup/pull/205)
[droolsjbpm-knowledge](https://github.com/kiegroup/droolsjbpm-knowledge/pull/521)
[drools](https://github.com/kiegroup/drools/pull/3628)
[optaplanner](https://github.com/kiegroup/optaplanner/pull/1333)
[appformer](https://github.com/kiegroup/appformer/pull/1143)
[jbpm](https://github.com/kiegroup/jbpm/pull/1917)
[kie-jpmml-integration](https://github.com/kiegroup/kie-jpmml-integration/pull/54)
[jbpm-wb](https://github.com/kiegroup/jbpm-wb/pull/1489)

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by mbiarnes** [comment]
Jenkins retest this

**Comment by mbiarnes** [comment]
Jenkins retest this

File: jbpm-installer/src/main/resources/build.properties [hunk 1]
```diff
@@ -27,11 +27,11 @@ jBPM.url=https://origin-repository.jboss.org/nexus/service/local/artifact/maven/
...
 # for WildFly19:
-#   jBPM.console.url=https://origin-repository.jboss.org/nexus/content/groups/public-jboss/org/kie/business-central/${jBPM.version}/business-central-${jBPM.version}-wildfly19.war
+#   jBPM.console.url=https://origin-repository.jboss.org/nexus/content/groups/public-jboss/org/kie/business-central/${jBPM.version}/business-central-${jBPM.version}-wildfly23.war
 # or:
 #<SNAPSHOT>
-#   jBPM.console.url=https://origin-repository.jboss.org/nexus/service/local/artifact/maven/redirect?r=snapshots&g=org.kie&a=business-central&v=${jBPM.version}&e=war&c=wildfly19
-jBPM.console.url=https://origin-repository.jboss.org/nexus/service/local/artifact/maven/redirect?r=snapshots&g=org.kie&a=business-central&v=${jBPM.version}&e=war&c=wildfly19
+#   jBPM.console.url=https://origin-repository.jboss.org/nexus/service/local/artifact/maven/redirect?r=snapshots&g=org.kie&a=business-central&v=${jBPM.version}&e=war&c=wildfly23
+jBPM.console.url=https://origin-repository.jboss.org/nexus/service/local/artifact/maven/redirect?r=snapshots&g=org.kie&a=business-central&v=${jBPM.version}&e=war&c=wildfly23
```
Comment (mareknovotny) [inline jbpm-installer/src/main/resources/build.properties:34]: we should replace origin repository whenever it is still referenced I think

File: jbpm-installer/src/main/resources/build.properties [hunk 2]
```diff
@@ -40,11 +40,11 @@ jBPM.console.url=https://origin-repository.jboss.org/nexus/service/local/artifac
 # for EAP7:
 #   jBPM.casemgmt.url=https://origin-repository.jboss.org/nexus/content/groups/public-jboss/org/jbpm/jbpm-wb-case-mgmt-showcase/${jBPM.version}/jbpm-wb-case-mgmt-showcase-${jBPM.version}-eap7.war
 # for WildFly19:
-#   jBPM.casemgmt.url=https://origin-repository.jboss.org/nexus/content/groups/public-jboss/org/jbpm/jbpm-wb-case-mgmt-showcase/${jBPM.version}/jbpm-wb-case-mgmt-showcase-${jBPM.version}-wildfly19.war
+#   jBPM.casemgmt.url=https://origin-repository.jboss.org/nexus/content/groups/public-jboss/org/jbpm/jbpm-wb-case-mgmt-showcase/${jBPM.version}/jbpm-wb-case-mgmt-showcase-${jBPM.version}-wildfly23.war
```
Comment (mareknovotny) [inline jbpm-installer/src/main/resources/build.properties:43]: we should replace origin repository whenever it is still referenced I think

File: jbpm-installer/src/main/resources/build.properties [hunk 3]
```diff
@@ -40,11 +40,11 @@ jBPM.console.url=https://origin-repository.jboss.org/nexus/service/local/artifac
...
 # or:
 #<SNAPSHOT>
-#   jBPM.casemgmt.url=https://origin-repository.jboss.org/nexus/service/local/artifact/maven/redirect?r=snapshots&g=org.jbpm&a=jbpm-wb-case-mgmt-showcase&v=${jBPM.version}&e=war&c=wildfly19
-jBPM.casemgmt.url=https://origin-repository.jboss.org/nexus/service/local/artifact/maven/redirect?r=snapshots&g=org.jbpm&a=jbpm-wb-case-mgmt-showcase&v=${jBPM.version}&e=war&c=wildfly19
+#   jBPM.casemgmt.url=https://origin-repository.jboss.org/nexus/service/local/artifact/maven/redirect?r=snapshots&g=org.jbpm&a=jbpm-wb-case-mgmt-showcase&v=${jBPM.version}&e=war&c=wildfly23
+jBPM.casemgmt.url=https://origin-repository.jboss.org/nexus/service/local/artifact/maven/redirect?r=snapshots&g=org.jbpm&a=jbpm-wb-case-mgmt-showcase&v=${jBPM.version}&e=war&c=wildfly23
```
Comment (mareknovotny) [inline jbpm-installer/src/main/resources/build.properties:47]: we should replace origin repository whenever it is still referenced I think

**Comment by mareknovotny** [comment]
@mbiarnes you need to also replace all `<artifactId>cargo-maven2-plugin</artifactId>` with `<artifactId>cargo-maven3-plugin</artifactId>` ;)

**Review by mareknovotny — CHANGES_REQUESTED** [review]
replace `<artifactId>cargo-maven2-plugin</artifactId>` to `<artifactId>cargo-maven3-plugin</artifactId>`

**Comment by mbiarnes** [comment]
Jenkins retest this

**Review by mareknovotny — APPROVED** [review]

**Comment by mbiarnes** [comment]
Jenkins retest this

**Comment by mbiarnes** [comment]
Jenkins execute fdb

File: jbpm-installer/src/main/resources/build.properties [hunk 4]
```diff
@@ -93,7 +93,7 @@ eclipse.clean.workspace=true
 #   jboss.server.deploy.dir=${jboss.home}/standalone/deployments
 #   jboss.server.data.dir=${jboss.home}/standalone/data
 #   jboss.clean.repository=true
-jboss.server.wildfly.version=19.1.0.Final
+jboss.server.wildfly.version=23.0.1.Final
```
Comment (mareknovotny) [inline jbpm-installer/src/main/resources/build.properties:96]:
```suggestion
jboss.server.wildfly.version=23.0.2.Final
```

File: jbpm-human-task/jbpm-human-task-audit/pom.xml [hunk 5]
```diff
@@ -71,12 +71,6 @@
           <groupId>jakarta.activation</groupId>
           <artifactId>jakarta.activation-api</artifactId>
         </exclusion>
-        <!-- collides with  javax.xml.bind:jaxb-api:jar:2.3.1:compile -->
-        <exclusion>
-          <groupId>jakarta.xml.bind</groupId>
-          <artifactId>jakarta.xml.bind-api</artifactId>
-        </exclusion>
-        <!-- collides with  org.jboss.spec.javax.xml.bind:jboss-jaxb-api_2.3_spec:jar:1.0.1.Final:compile -->
```
Comment (mareknovotny) [inline jbpm-human-task/jbpm-human-task-audit/pom.xml:79]: i guess you removed this comment by accident? this should stay or change to something which explains why the exclusion was added here

File: jbpm-human-task/jbpm-human-task-jpa/pom.xml [hunk 6]
```diff
@@ -77,26 +77,20 @@
           <groupId>jakarta.activation</groupId>
           <artifactId>jakarta.activation-api</artifactId>
         </exclusion>
-        <!-- collides with  javax.xml.bind:jaxb-api:jar:2.3.1:compile -->
-        <exclusion>
-          <groupId>jakarta.xml.bind</groupId>
-          <artifactId>jakarta.xml.bind-api</artifactId>
-        </exclusion>
-        <!-- collides with  org.jboss.spec.javax.xml.bind:jboss-jaxb-api_2.3_spec:jar:1.0.1.Final:compile -->
```
Comment (mareknovotny) [inline jbpm-human-task/jbpm-human-task-jpa/pom.xml:85]: again this should stay or change the comment content  to something which explains why the exclusion was added here

File: jbpm-installer/src/main/resources/build.properties (hunk 1 above)
Comment (mbiarnes) [reply jbpm-installer/src/main/resources/build.properties]: DONE

File: jbpm-installer/src/main/resources/build.properties (hunk 2 above)
Comment (mbiarnes) [reply jbpm-installer/src/main/resources/build.properties]: DONE

File: jbpm-installer/src/main/resources/build.properties (hunk 3 above)
Comment (mbiarnes) [reply jbpm-installer/src/main/resources/build.properties]: DONE

File: jbpm-human-task/jbpm-human-task-audit/pom.xml (hunk 5 above)
Comment (mbiarnes) [reply jbpm-human-task/jbpm-human-task-audit/pom.xml:79]: DONE

File: jbpm-human-task/jbpm-human-task-jpa/pom.xml (hunk 6 above)
Comment (mbiarnes) [reply jbpm-human-task/jbpm-human-task-jpa/pom.xml:85]: DONE

**Comment by mbiarnes** [comment]
Jenkins retest this

**Comment by mbiarnes** [comment]
Jenkins retest this

**Comment by mbiarnes** [comment]
Jenkins retest this

**Comment by mbiarnes** [comment]
@mareknovotny  @fjtirado the build was SUCCESSFUL but one [test](https://eng-jenkins-csb-business-automation.apps.ocp4.prod.psi.redhat.com/job/KIE/job/master/job/pullrequest/job/jbpm-master.pr/161/) failed

**Comment by mareknovotny** [comment]
@mbiarnes that failure is i hope only with the environment problem `BindException(message='Address already in use (Bind failed)'`
so there were some process blocking the network address. Probably not correctly stopped previous WF container-

**Comment by mbiarnes** [comment]
This PR will be closed since Tiago Bento and William Siqueira are working on the same issue:
https://github.com/kiegroup/jbpm/pull/1974

**Pull request closed without merging**